### PR TITLE
Change insights to use vscode splitview

### PR DESCRIPTION
**Pull request by anthonydresser**
image

Proof of concept for using the vscode splitview in our split views. I can convert the rest if we want to.

### CODE REVIEW

File: src/sql/parts/insights/browser/insightsDialogView.ts [hunk 1]
```diff
@@ -211,8 +213,13 @@ export class InsightsDialogView extends Modal {
 			});
 		}));
 
-		this._splitView.addView(topTableView);
-		this._splitView.addView(bottomTableView);
+		attachPanelStyler(topTableView, this._themeService);
+		attachPanelStyler(bottomTableView, this._themeService);
+
+		panelCreation.then(() => {
+			this.panelView.addPanel(topTableView, 1, 0);
```
Comment (AbbiePetcht) [inline src/sql/parts/insights/browser/insightsDialogView.ts]: >1 [](start = 41, length = 1)

Why size is 1 here?

File: src/sql/base/browser/ui/modal/modal.ts [hunk 2]
```diff
@@ -305,17 +305,21 @@ export abstract class Modal extends Disposable implements IThemable {
...
 		TelemetryUtils.addTelemetry(this._telemetryService, TelemetryKeys.ModalDialogOpened, { name: this._name });
 	}
 
 	/**
 	 * Required to be implemented so that scrolling and other functions operate correctly. Should re-layout controls in the modal
 	 */
-	protected abstract layout(height?: number): void;
+	protected abstract layout(dimension: Dimension): void;
+
+	protected _layout(): void {
```
Comment (AbbiePetcht) [inline src/sql/base/browser/ui/modal/modal.ts]: >_layout [](start = 11, length = 7)

Should rename this function. _layout and layout is confusing especially calling from another class. Maybe LayoutWithBodyContent?

File: src/sql/base/browser/ui/modal/modal.ts [hunk 3]
```diff
@@ -305,17 +305,21 @@ export abstract class Modal extends Disposable implements IThemable {
...
 
-		this.layout(DOM.getTotalHeight(this._builder.getHTMLElement()));
+		this._layout();
 		TelemetryUtils.addTelemetry(this._telemetryService, TelemetryKeys.ModalDialogOpened, { name: this._name });
 	}
 
 	/**
 	 * Required to be implemented so that scrolling and other functions operate correctly. Should re-layout controls in the modal
 	 */
-	protected abstract layout(height?: number): void;
```
Comment (AbbiePetcht) [inline src/sql/base/browser/ui/modal/modal.ts]: >layout [](start = 20, length = 6)

When should be call layout VS _layout

Comment (anthonydresser) [reply src/sql/base/browser/ui/modal/modal.ts]: _layout is basically the internal method to tell the modal to layout itself. It gets the sizes and then calls layout() with the correct dimension.

File: src/sql/base/browser/ui/modal/modal.ts (hunk 2 above)
Comment (anthonydresser) [reply src/sql/base/browser/ui/modal/modal.ts]: the layout(dimension) method falls in line with vscode's ui elements of having a layout method with a measurement. the _layout was basically just a helper function that grabs all the measurements for the modal and calls the correct layout method

File: src/sql/parts/insights/browser/insightsDialogView.ts (hunk 1 above)
Comment (anthonydresser) [reply src/sql/parts/insights/browser/insightsDialogView.ts]: I think this size is a relative size value, not a pixel size

**Comment by kburtram** [comment]
@anthonydresser is this targeting June release?  If so, could you please try to merge today?

**Comment by kburtram** [comment]
@anthonydresser thanks for confirming!

**Comment by kburtram** [comment]
@anthonydresser do we want to take this PR?  Or should we close it out?

**Comment by anthonydresser** [comment]
I'm going to close this since there are still some issues

**Pull request closed without merging**